### PR TITLE
chore!: bump to new @waku

### PR DESCRIPTION
**Pull request by weboko**
Adds usage of: @waku/interfaces & @waku/utils
Update version to 0.0.14

### CODE REVIEW

File: src/codec.spec.ts [hunk 1]
```diff
@@ -1,20 +1,22 @@
-import { expect } from "chai";
+import {
+  DecodedMessage,
```
Comment (fryorcraken) [inline src/codec.spec.ts:2]: nitpick: This might be possible to `import type`.

File: src/codec.spec.ts [hunk 2]
```diff
@@ -34,22 +36,22 @@ describe("RLN codec with version 0", () => {
     rlnInstance.insertMember(memKeys.IDCommitment);
 
     const rlnEncoder = new RLNEncoder(
-      new EncoderV0(TestContentTopic),
+      new Encoder(TestContentTopic),
```
Comment (fryorcraken) [inline src/codec.spec.ts:39]: nitpick: `createEncoder` is preferred as it is the API we encourage developers to use.

File: src/codec.spec.ts [hunk 3]
```diff
@@ -132,7 +138,7 @@ describe("RLN codec with version 1", () => {
     const protoResult = await rlnDecoder.fromWireToProtoObj(bytes!);
 
     expect(protoResult).to.not.be.undefined;
-    const msg = (await rlnDecoder.fromProtoObj(protoResult!))!;
+    const msg = (await rlnDecoder.fromProtoObj("", protoResult!))!;
```
Comment (fryorcraken) [inline src/codec.spec.ts:141]: nitpick: might be good to justify the use of an empty string. Maybe a glob const `DUMMY_CONTENT_TOPIC`.

File: src/message.ts [hunk 4]
```diff
@@ -1,41 +1,51 @@
...
-  const contentTopicBytes = utils.utf8ToBytes(msg.contentTopic ?? "");
+export function toRLNSignal(contentTopic: string, msg: IMessage): Uint8Array {
+  const contentTopicBytes = utils.utf8ToBytes(contentTopic ?? "");
   return new Uint8Array([...(msg.payload ?? []), ...contentTopicBytes]);
 }
 
-export class RlnMessage<T extends Message> implements Message {
+export class RlnMessage<T extends IDecodedMessage> implements IDecodedMessage {
+  public pubSubTopic = "";
+  public ephemeral = false;
```
Comment (fryorcraken) [inline src/message.ts:18]: Setup a getter, this returns an incorrect value.

**Review by fryorcraken — CHANGES_REQUESTED** [review]
Request changes: `DecodedMessage.ephemeral` does not return the expected value afaik

File: src/codec.spec.ts (hunk 3 above)
Comment (weboko) [reply src/codec.spec.ts:141]: Good one, agree!

**Review by fryorcraken — APPROVED** [review]

**Pull request merged**